lib/client.js

```diff
@@ -120,6 +121,8 @@ function Client (broker, conn, req) {
   }
 
   this.deliverQoS = function deliverQoS (_packet, cb) {
+    _packet.topic = that.broker.persistence.restoreOriginalTopicFromSharedOne(_packet.topic)
```

Comment:
Same as above, could omit the call in case shared topics are not enabled or topic doesn't start with `$shared`?